Minor cleanup of ThriftTransportPool

ctubbsii opened the pull request:
* Remove unnecessary memoization of the checker thread
* Inline the Closer Runnable type and make it a lambda
* Use primitive long for `ERROR_THRESHOLD`
* Reorder all the class members before all the inner-classes, and place
  the fields above the methods to make it easier to navigate the class
* Make constructor private and use a static method to construct and
  start the checker thread after construction, then return the instance
* Make ThriftTransportPool immutable by passing in the maximum age of
  idle transports, rather than setting it after it is constructed; this
  also changes the default to the general purpose RPC timeout config
  value for all servers, rather than just the manager; clients continue
  to use the hard-coded 3 seconds, just as before this change

ctubbsii commented:
This is the follow-up from the discussion in https://github.com/apache/accumulo/pull/2590#discussion_r835559945

dlmarion commented on core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java:
```diff
@@ -1014,11 +1014,14 @@ public ZooReader getZooReader() {
 
+  protected long getTransportPoolMaxAgeMillis() {
+    return 3000; // 3 seconds for clients
+  }
```
I think this is great. In fact, if we can make it configurable via a client property (ClientProperty.GENERAL_RPC_TIMEOUT ?), then we can likely close [ACCUMULO-2069](https://issues.apache.org/jira/browse/ACCUMULO-2069). I remember when I submitted that issue that I did some testing and the overhead of closing and recreating connections was significant.

ctubbsii replied:
I can definitely change this so it uses the normal `Property.GENERAL_RPC_TIMEOUT`. I didn't do it before because we have a default value of 2 minutes for our general RPC timeouts, and that seemed to be very far from the 3 second hard-coded value previously used. So, I went with a change that left the current value in place.

However, I think it's probably okay to increase it to 2 minutes by default, after reading the comment on [ACCUMULO-2069 (comment)](https://issues.apache.org/jira/browse/ACCUMULO-2069?focusedCommentId=13861091&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13861091). I think the value of 3 seconds was probably used so clients were more responsive in the absence of a close method. However, that is no longer necessary, since there is a proper close method on clients now, and all the transports get closed when the client is closed. So, it's okay that they are left around longer.

ctubbsii replied:
Addressed in 410bd68ce668287f4a5f18d313318bdf899e5927

dlmarion reviewed (changes requested):
Looks great, just one comment about making the client RPC timeout configurable to close an old issue.

ctubbsii commented:
GH Actions test seems like it was interrupted. Not sure what's going on. Doesn't seem related to my code change... but possibly the result of the increased timeout. I'll rerun in GH Actions. If it doesn't work still, I'll do a full IT build and try to track down which tests are sensitive to the idle transport timeout default change.

ctubbsii commented:
Okay, I fixed the issue with the test breaking. The problem was that the test timed out while waiting for the checker thread to terminate, which it wasn't going to do quickly, because I had the loop time based on the max age time. I changed it in f96d34e62c6bc8aba087af3c12df2d943c10cbc1 so that it loops more frequently, so it can more easily detect when shutdown has occurred and can exit, but it doesn't run the check every loop... it only runs the check at intervals half way to the max age, but no more frequently than 4 times a second, and no less frequently than once a minute.
